wezterm/src/main.rs

```diff
@@ -728,8 +768,18 @@ async fn run_cli_async(config: config::ConfigHandle, cli: CliCommand) -> anyhow:
             smol::future::pending().await
         }
         CliSubCommand::TlsCreds => {
+            let out = std::io::stdout();
             let creds = client.get_tls_creds().await?;
-            codec::Pdu::GetTlsCredsResponse(creds).encode(std::io::stdout().lock(), 0)?;
+
+            match cli.output_format {
+                CliOutputFormat::Json => {
+                    serde_json::to_writer_pretty(out.lock(), &creds)?;
```

Comment:
I think this should be calling `encode` of some sort and then serializing the result as `Vec<u8>` instead probably.

Reply:
json is always encoded as utf8 output and that byte sequence is written to the writer, so no additional encoding is needed

Reply:
However, I'm not sure if this particular command should have json output given that it is an implementation detail of bootstrapping a mux connection!

Reply:
Cool, I think then it makes sense to pull --output-format out into each CliSubCommand so it'd be `wezterm cli list --output-format json`?

I'll do a pass at this later, after peeking through the testsuite results.